### PR TITLE
Cleanup the spec file

### PR DESCRIPTION
**Pull request by scarabeusiv**
As per commits first I detect libxml properly so we can build without hassle on Fedora.
I also by default told it to link with -fPIC and disable static libraries.

Then I cleanup the spec with spec-cleaner. It still builds on Fedora_24 and all the distributions enabled in the develprj like before.
Most notable change is that now the libstorage@LIBVERSION@ is able to be coinstalled with multiple versions. Translations were split to lang subpackage.

### CODE REVIEW

**Comment by aschnell** [comment]
Closing since there was no reply from requester.

**Comment by aschnell** [comment]
My review request was still pending and AFAIS thus not visible to others, sorry for that.

**Comment by scarabeusiv** [comment]
Hopefully addressed the comment above.

FWIW The boost regex library could be also detected in configure.ac but as it is now at least erroring out if user ommits the headers by accident it should be quite okayish.

**Comment by aschnell** [comment]
Fails in factory with 'ld: cannot find -lboost_unit_test_framework'. I suppose you need to add libboost_test-devel.

File: libstorage-ng.spec.in [hunk 1]
```diff
@@ -15,246 +16,182 @@
...
-Requires:	lvm2
-Requires:	mdadm >= 3.3
-Requires:	multipath-tools
-Requires:	parted >= 3.1
-Requires:	udev
-Requires:	util-linux >= 2.16
+%package -n %{libname}
+Summary:        Library for storage management
+Group:          System/Libraries
+Requires:       %{name}-lang
```
Comment (aschnell) [inline libstorage-ng.spec.in]: Why not require the lang subpackage in the exact version?

Comment (scarabeusiv) [inline libstorage-ng.spec.in]: This is usually not done as it was considered fine to just pull the language, of course if you want just add the = %{version}

Comment (aschnell) [inline libstorage-ng.spec.in]: OK, I will likely add it someday.

**Comment by aschnell** [comment]
LGTM

**Pull request merged**